### PR TITLE
Bluetooth: Host: Fix missing pull left mem in scan pdu

### PR DESCRIPTION
**Pull request by LingaoM**
For num_reports field bigger 1, should pull left mem.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

File: subsys/bluetooth/host/scan.c [hunk 1]
```diff
@@ -606,6 +606,7 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 			 */
 			create_ext_adv_info(evt, &scan_info);
 			le_adv_recv(&evt->addr, &scan_info, &buf->b, evt->length);
+			(void)net_buf_pull_mem(buf, evt->length);
```
Comment (jori-nordic) [inline subsys/bluetooth/host/scan.c:609]: I see we're adding that line to almost all branches. I think it'd be better to have it after L598 if possible. What do you think @LingaoM ?

Comment (LingaoM) [reply subsys/bluetooth/host/scan.c:609]: In fact, it will not be easy. This will modify `buf->data` in advance, resulting in subsequent logic changes.

Comment (LingaoM) [reply subsys/bluetooth/host/scan.c]: Use `goto cont` may better.

**Review by jori-nordic — APPROVED** [review]

**Comment by jori-nordic** [comment]
I'm also working with `scan.c` lately (an attempt to fix #50786). I've seen that neither the zephyr LL or the softdevice controller enqueue multiple adv reports in one event. What controller are you using that does so? I'd like to also test with it.

**Comment by LingaoM** [comment]
> I'm also working with `scan.c` lately (an attempt to fix #50786). I've seen that neither the zephyr LL or the softdevice controller enqueue multiple adv reports in one event. What controller are you using that does so? I'd like to also test with it.

hah, i‘m use Realtek 8262 Serial Controller, Their behavior is indeed strange, but it is in line with the standard.

**Comment by LingaoM** [comment]
image

**Comment by jori-nordic** [comment]
> > I'm also working with `scan.c` lately (an attempt to fix #50786). I've seen that neither the zephyr LL or the softdevice controller enqueue multiple adv reports in one event. What controller are you using that does so? I'd like to also test with it.
> 
> hah, i‘m use Realtek 8262 Serial Controller, Their behavior is indeed strange, but it is in line with the standard.

Yes it's standards compliant. Could you check just one last thing? Does it interleave fragmented advertising reports.
You can use the `zephyr/samples/bluetooth/broadcaster_multiple` sample from https://github.com/jori-nordic/zephyr/tree/ext-adv-debug , I modified it so it should send 3 advertising PDUs, which would mean 3 adv reports on the host. If you flash it to like 3 devices, we will see if the controller tries to interleave reports (which we don't support rn).

**Comment by LingaoM** [comment]
I'm sorry, I can only test the current mr, I can't test the test you proposed for the time being.

**Comment by jori-nordic** [comment]
> I'm sorry, I can only test the current mr, I can't test the test you proposed for the time being.

yep no problem, thanks for the bugfix!

**Review by jhedberg — APPROVED** [review]
Looks fine, but I think you might be able to avoid the goto by converting the while look to a for loop. Anyway, this is fine as a quick fix too.

**Pull request merged**